NitroxServer/Program.cs

```diff
@@ -31,11 +31,15 @@ static void Main(string[] args)
                 return;
             }
 
-            ConsoleCommandProcessor CmdProcessor = NitroxServiceLocator.LocateService<ConsoleCommandProcessor>();
-            while (server.IsRunning)
-            {
-                CmdProcessor.ProcessCommand(Console.ReadLine());
+            if (Environment.UserInteractive)
+            {
+                ConsoleCommandProcessor CmdProcessor = NitroxServiceLocator.LocateService<ConsoleCommandProcessor>();
+                while (server.IsRunning)
+                {
+                    CmdProcessor.ProcessCommand(Console.ReadLine());
+                }

```

Comment:
Won't the application immediately close if it isn't stopped by the while loop?

Reply:
Apparently not, I thought that too. But since it's running in the background with other threads, it works. My guess is the main program must wait for other threads to terminate?

Reply:
I can now confirm that the thread which executes Listen in NitroxServer/Communication/UdpServer.cs (thread), runs in the foreground and hence ensures the application does not terminate.
https://docs.microsoft.com/en-us/dotnet/api/system.threading.thread?view=netframework-4.7.2#foreground-and-background-threads

I have also added a few more things to this pull request.